src/libraries/System.Security.Cryptography.Encoding/tests/Cbor/CborReader.String.cs

```diff
@@ -0,0 +1,35 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Buffers.Binary;
+
+namespace System.Security.Cryptography.Encoding.Tests.Cbor
+{
+    internal ref partial struct CborValueReader
+    {
+        private static readonly System.Text.Encoding s_utf8Encoding = new System.Text.UTF8Encoding(encoderShouldEmitUTF8Identifier: false, throwOnInvalidBytes: true);
+
+        // Implements major type 2 decoding per https://tools.ietf.org/html/rfc7049#section-2.1
+        public byte[] ReadByteString()
+        {
+            CborDataItem header = ReadInitialByte(expectedType: CborMajorType.ByteString);
+            int length = checked((int)ReadUnsignedInteger(header, out int additionalBytes));
```

Comment:
Newbie in C# question: You're reading an unsigned integer but assigning to a signed integer. Would that cause issues with the allocation below?

Reply:
You mean because of overflows? The `checked` keyword ensures that an exception is thrown if that happens.